src/parser.rs

```diff
@@ -178,6 +178,31 @@ pub trait Parser<I, O, E> {
         Value::new(self, val)
     }
 
+    /// Produce a value by calling the provided generator function
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # use winnow::{error::ErrMode,error::ErrorKind, error::InputError, Parser};
+    /// use winnow::ascii::alpha1;
+    /// # fn main() {
+    ///
+    /// let mut parser = alpha1.value_with(|| 1234);
```

Comment:
What separates this from the following?
```suggestion
    /// let mut parser = alpha1.map(|_| 1234);
```

Reply:
I've got nothing to say except—yeah, that's a very good point. Closing :^)